src/patternfly/components/Popover/styles.scss

```diff
@@ -0,0 +1,184 @@
+  --pf-c-popover__content--PaddingBottom: var(--pf-global--spacer--lg);
+  --pf-c-popover__content--PaddingLeft: var(--pf-global--spacer--lg);
+  --pf-c-popover__content--FontSize: var(--pf-global--FontSize--md);
+  --pf-c-popover__content--Color: var(--pf-global--Color--100);
+  --pf-c-popover__content--BackgroundColor: var(--pf-global--BackgroundColor--100);
+
+  // Header variables
+  --pf-c-popover__header--MarginTop: var(--pf-global--spacer--xs);
+  --pf-c-popover__header--MarginBottom: var(--pf-global--spacer--sm);
+
+  // HeaderText variables
+  --pf-c-popover__header-text--PaddingTop: var(--pf-global--spacer--xxs);
+  --pf-c-popover__header-text--FontSize: var(--pf-global--FontSize--xl);
+
+  // Close
+  --pf-c-popover__close--PaddingLeft: var(--pf-global--spacer--sm);
+  --pf-c-popover__close--FontSize: var(--pf-global--FontSize--xl);
+
+  // Body variables
+  --pf-c-popover__body--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-popover__body--FontSize: var(--pf-global--FontSize--md);
+
+  position: relative;
+  display: inline-block;
+
+  // Elements
+  &__container {
+    // This element always needs to be light
+    --pf-c-popover__content--Color: var(--pf-global--Color--dark-100);
+    @extend %pf-t-light;
+
+    position: absolute;
+    z-index: var(--pf-c-popover__container--ZIndex);
+    display: flex;
+    width: var(--pf-c-popover__container--Width);
+    filter: var(--pf-c-popover__arrow--DropShadow);
+  }
+
+  &__arrow {
+    position: absolute;
+    border-color: var(--pf-c-popover__arrow--BorderColor);
+    border-style: solid;
+  }
+
+  &__content {
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+    min-width: var(--pf-c-popover__content--MinWidth);
+    padding-top: var(--pf-c-popover__content--PaddingTop);
+    padding-right: var(--pf-c-popover__content--PaddingRight);
+    padding-bottom: var(--pf-c-popover__content--PaddingBottom);
+    padding-left: var(--pf-c-popover__content--PaddingLeft);
+    font-size: var(--pf-c-popover__content--FontSize);
+    color: var(--pf-c-popover__content--Color);
+    background-color: var(--pf-c-popover__content--BackgroundColor);
+  }
+
+  &__header {
+    display: flex;
+    align-items: flex-start;
+    justify-content: space-between;
+    margin-top: var(--pf-c-popover__header--MarginTop);
+    margin-bottom: var(--pf-c-popover__header--MarginBottom);
+  }
+
+  &__header-text {
+    flex: 1;
+    padding-top: var(--pf-c-popover__header-text--PaddingTop);
+    font-size: var(--pf-c-popover__header-text--FontSize);
+  }
+
+  &__close {
+    padding-left: var(--pf-c-popover__close--PaddingLeft);
+
+    .pf-c-button.pf-m-action {
+      font-size: var(--pf-c-popover__close--FontSize);
+    }
+  }
+
+  &__body {
+    padding-right: var(--pf-c-popover__body--PaddingRight);
+    font-size: var(--pf-c-popover__body--FontSize);
+  }
+
+  // Modifiers
+  &.pf-m-left &__container {
+    top: 50%;
```

Comment:
I don't think this positioning makes sense to be in here.  Since frameworks/JS will be responsible for positioning these elements I think it would make more sense to leave it out since it is just one more thing that we will need to overwrite. The modifiers applied for left, right, etc should only drive the caret. Most frameworks, including css/js only frameworks, will append the tooltip/popover to the body instead of keeping them inline with the target.  This will allow them to be placed in areas that have overflow set to hidden.

Reply:
Wouldn't most popovers act like modals in terms of needing an explicit action to open the popover via keyboard and once open capturing the keyboard focus and only returning the focus to the control when it closes? I don't know if the user should be forced into navigating through a popover if they don't want to.

Tooltips would display non-interactive text and therefore no focus is needed at all. Would aria-describedby be sufficient with a hidden node containing the tooltip text?

To address @dmiller9911's point. I think that for no JS tooltip/popover the current implementation makes sense. But when they are appended to the body and used with a lib (such as popperjs), then we would need to override the positioning and transforms. It's easily done but maybe additional modifiers that could be applied directly to the arrow to change it's orientation but not apply any positioning.

Reply:
@christianvogt I am good with overriding is that is what is decided.  It would be nice to remove the cascade here though with the child selector.  This will either require us to use `!important` or make our own selector that targets the child.  Further complicating the override. This could be done by removing `container` and moving all the styles up to `pf-c-tooltip`.

Reply:
@dmiller9911 A lib like popperjs would apply the position via styles directly on the element which would override any css rules. I doubt you'd have to create new rules for your use case. But any positioning provided by pf would not be necessary with popperjs.

Reply:
> Wouldn't most popovers act like modals in terms of needing an explicit action to open the popover via keyboard and once open capturing the keyboard focus and only returning the focus to the control when it closes? 

Copying @mcarrano on this one. I _think_ popovers would be modal and not modeless, but I want to make sure designers are on the same page. If modal, then the issue of managing focus is probably easier to solve if we go with @dmiller9911's approach with the DOM.

> I don't know if the user should be forced into navigating through a popover if they don't want to.

I agree, I expect the popover to only display based on explicit action by a user. 

> Tooltips would display non-interactive text and therefore no focus is needed at all. Would aria-describedby be sufficient with a hidden node containing the tooltip text?

Yes, I agree with this one too. But related to accessibility of tooltips, there are a couple of points worth noting:

- In previous screen reader tests, I've noticed that if I had an aria attribute reference the id of a `hidden` element, the screen reader ignored it. So maybe it would be "hidden" using something like a sr-only class, but not `hidden` with the html attribute. 
  - @jlmitch5, @LHinson  I'm wondering if it makes sense to capture accessibility concerns related to tooltips in a separate issue. It might be worth setting up some test pages with variations to see what works and what doesn't in the context of a screen reader.
- There's a [new WCAG 2.1 item (1.4.13)](https://www.w3.org/WAI/WCAG21/quickref/#content-on-hover-or-focus) that affects tooltips, where anything that becomes visible on hover or focus must be dismissible via the Esc key

Reply:
Yes, the popover behaves like a modal. The user takes an explicit action to open the popover and it is closed either by a Close button inside the popover or by clicking away from it.

Reply:
@dmiller9911 unfortunately, I can't excise the container dom element, because it is being used to separate the tooltip's content from the parents which is wrapped by the component block element.

Reply:
That is ok. We will work around anything that we have to. It might work as is too. We will know more once we start implementing it.